tpm2/test/tpm2_test.go

```diff
@@ -700,6 +716,119 @@ func TestSign(t *testing.T) {
 	})
 }
 
+func TestSignWithAttestationKey(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+	run := func(t *testing.T, data []byte, wantErr bool, pub Public) {
+		signerHandle, signerPub, err := CreatePrimary(rw, HandleOwner, pcrSelection7, emptyPassword, defaultPassword, pub)
+		if err != nil {
+			t.Fatalf("CreatePrimary failed: %v", err)
+		}
+		defer FlushContext(rw, signerHandle)
+
+		var scheme *SigScheme
+		if pub.RSAParameters != nil {
+			scheme = pub.RSAParameters.Sign
+		}
+		if pub.ECCParameters != nil {
+			scheme = pub.ECCParameters.Sign
+		}
```

Comment:
Could we just use a `nil` scheme here to make things more simple?

Reply:
Good catch, agreed!

Reply:
I think making this scheme `nil` got missed.

Reply:
whoops, fixed one test but not the other. Sorry about this!